python/algos/multitask/mmoe/mmoe_agent.py

```diff
@@ -0,0 +1,218 @@
+#
+# Copyright 2022 DMetaSoul
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import torch
+import metaspore as ms
+import torch.nn.functional as F
+import json
+from datetime import datetime
+import struct
+import numpy as np
+import pandas as pd
+from pyspark.sql.functions import col
+
+from metaspore._metaspore import Message
+
+class MMoEAgent(ms.PyTorchAgent):
+    '''
+    def nansum(self, x):
```

Comment:
should we delete these commented code?

Reply:
We should keep these code for customizing 'compute_loss' function.